src/v/compat/raft_generator.h

```diff
@@ -0,0 +1,172 @@
+/*
```

Comment:
Maybe put the actual test implementations in a separate directory so we have a clean split between "compat framework" vs "code that uses the compat framework to implement test code".

Reply:
+1 will do

Reply:
done

Reply:
for me it looks like `check.h`, `generator.h`, `run.h`, `run.cc` and `json.h` (the framework) are still in the same directory as the actual tests (`metadata_dissemination_*`, `raft_*`).